ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1375,6 +1471,16 @@ protected void maybeProcessCallbacks() {
     @Override
     void handleBody(InputStream payload, MessageMetadata messageMetadata, MessageInfo messageInfo)
         throws IOException, MessageFormatException {
+
+      if (isMetadataCacheHit()) {
+        getOptions().ageAtAccessTracker.trackAgeAtAccess(serverBlobProperties.getCreationTimeInMs());
+        handleMetadataBlob(null, blobInfo.getUserMetadata(), null);
+        // Do not call updateTtlIfRequired(). If the blob expires, the frontend will find out from backend during the next GET request
+        // TODO: Deserialize payload and do basic validations to check if cached blob properties are correct.
+        // TODO: Should we decrypt ? Probably no.
```

Comment:
One thing I am not very sure of is what to do if the blobInfo we fetch to validate the cache entry is corrupted or not found. Do we invalidate the cache entry and retry to fetch the entire metadata from backend ? @justinlin-linkedin